File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>ump;
        for (int i = 0; i < nums.size() ; i++){
            int rem = target - nums[i];
            if (ump[rem]) return {ump[rem]-1,i};
            ump[nums[i]] = i+1;
        }
        return {-1,-1};
    }
};

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let mp = new Map();
    for (let i = 0 ; i < nums.length; i++){
        let diff = target - nums[i];
        if (mp.has(diff)){
            return [mp.get(diff),i];
        }
        mp.set(nums[i],i);
    }
    return [];
};